python/semantic_kernel/agents/agent.py

```diff
@@ -478,3 +560,336 @@ def as_mcp_server(
+    ) -> _TAgent:
+        """Create a single agent instance from a YAML string.
+
+        Args:
+            yaml_str: The YAML string defining the agent.
+            kernel: The Kernel instance to use for tool resolution and agent initialization.
+            settings: The settings to use for the agent.
+            extras: Additional parameters to resolve placeholders in the YAML.
+            **kwargs: Additional parameters passed to the agent constructor if required.
+
+        Returns:
+            An instance of the requested agent.
+
+        Raises:
+            AgentInitializationException: If the YAML is invalid or the agent type is not supported.
+
+        Example:
+            agent = await AgentRegistry.create_agent_from_yaml(
+                yaml_str, kernel=kernel, service=AzureChatCompletion(),
+            )
+        """
+        data = yaml.safe_load(yaml_str)
+
+        agent_type = data.get("type", "").lower()
+        if not agent_type:
+            raise AgentInitializationException("Missing 'type' field in agent definition.")
+
+        if agent_type not in AGENT_TYPE_REGISTRY:
+            raise AgentInitializationException(f"Agent type '{agent_type}' not registered.")
+
+        if not kernel:
+            kernel = Kernel()
+
+        agent_cls = AGENT_TYPE_REGISTRY[agent_type]
+
+        if not isinstance(agent_cls, DeclarativeSpecProtocol):
+            raise AgentInitializationException(
+                f"Agent class '{agent_cls.__name__}' does not support declarative spec loading."
+            )
+
+        yaml_str = agent_cls.resolve_placeholders(yaml_str, settings, extras)
+        data = yaml.safe_load(yaml_str)
+
+        return await agent_cls.from_dict(
+            data,
+            kernel=kernel,
+            settings=settings,
+            **kwargs,
+        )
+
+    @staticmethod
+    async def create_from_dict(
+        data: dict,
+        *,
+        kernel: Kernel,
+        settings: "KernelBaseSettings | None" = None,
+        **kwargs,
+    ) -> _TAgent:
+        """Create a single agent instance from a dictionary.
+
+        Args:
+            data: The dictionary defining the agent fields.
+            kernel: The Kernel instance to use for tool resolution and agent initialization.
+            settings: The settings to use for the agent.
+            **kwargs: Additional parameters passed to the agent constructor if required.
+
+        Returns:
+            An instance of the requested agent.
+
+        Raises:
+            AgentInitializationException: If the dictionary is missing a 'type' field or the agent type is unsupported.
+
+        Example:
+            agent = await AgentRegistry.create_agent_from_dict(agent_data, kernel=kernel)
+        """
+        agent_type = data.get("type", "").lower()
+
+        if not agent_type:
+            raise AgentInitializationException("Missing 'type' field in agent definition.")
+
+        if agent_type not in AGENT_TYPE_REGISTRY:
```

Comment:
one issue I see with this is that the AGENT_TYPE_REGISTRY is only filled when a agent is actually loaded, I think we should pre-load all the built-in agents and just use the register_agent_type for custom agent classes to allow extensibility. In my mind, one should be able to just declare the spec and without any additional imports run the agent...

Reply:
Yeah, valid concern and good callout. I'm adding some handling to make sure we have the agent types available without having to explicitly load the required them.